packages/admin/src/pages/build-application/[applicationId]/survey.page.tsx

```diff
@@ -1,70 +1,97 @@
 import { postSurveyResponse } from '../../../services/SatisfactionSurveyService';
-import { getSessionIdFromCookies } from '../../../utils/session';
+import InferProps from '../../../types/InferProps';
+import QuestionPageGetServerSideProps from '../../../utils/QuestionPageGetServerSideProps';
 import { getApplicationFormSummary } from '../../../services/ApplicationService';
 import { getGrantScheme } from '../../../services/SchemeService';
 import { errorPageRedirect } from './publish-service-errors';
+import Scheme from '../../../types/Scheme';
+
+type PageBodyResponse = {
+  satisfaction: string;
+  comment: string;
+};
 
-export const getServerSideProps: GetServerSideProps = async ({
-  query,
-  params,
-  req,
-}) => {
-  const { applicationId } = params as Record<string, string>;
+export async function getServerSideProps(context: GetServerSidePropsContext) {
+  const { applicationId } = context.params as Record<string, string>;
 
-  let grantScheme;
+  let grantScheme: Scheme;
   try {
     const applicationFormSummary = await getApplicationFormSummary(
       applicationId,
-      getSessionIdFromCookies(req)
+      getSessionIdFromCookies(context.req)
     );
     grantScheme = await getGrantScheme(
       applicationFormSummary.grantSchemeId,
-      getSessionIdFromCookies(req)
+      getSessionIdFromCookies(context.req)
     );
   } catch (err) {
     return errorPageRedirect(applicationId, true);
   }
-
   const schemeId = grantScheme.schemeId;
-  const sessionId = getSessionIdFromCookies(req);
-  const backendUrl = `${
-    getConfig().serverRuntimeConfig.backendHost
-  }/feedback/add`;
-  const fieldErrors: ValidationError[] = query.fieldErrors
-    ? [JSON.parse(query.fieldErrors as string)]
-    : [];
 
-  return {
-    props: {
-      backendUrl,
-      fieldErrors,
-      schemeId,
-      sessionId,
+  const satisfaction = context.query.satisfaction
+    ? decodeURIComponent(context.query.satisfaction as string)
+    : null;
+
+  const comment = context.query.comment
+    ? decodeURIComponent(context.query.comment as string)
+    : null;
+
+  const url = `${getConfig().serverRuntimeConfig.backendHost}/feedback/add`;
+
+  async function handleRequest(body: PageBodyResponse, jwt: string) {
+    await postSurveyResponse(
+      body.satisfaction,
+      body.comment,
+      jwt,
+      url,
+      'application'
+    );
+    return body.satisfaction;
+  }
+
+  async function fetchPageData() {
+    return {
+      satisfaction: satisfaction,
+      comment: comment,
+      journey: 'application',
+    };
+  }
+
+  return QuestionPageGetServerSideProps<
+    PageBodyResponse,
+    Awaited<ReturnType<typeof fetchPageData>>,
+    Awaited<ReturnType<typeof handleRequest>>
+  >({
+    context,
+    fetchPageData,
+    handleRequest,
+    jwt: getSessionIdFromCookies(context.req),
+    onErrorMessage: 'Failed to send feedback.',
+    onSuccessRedirectHref: () => {
+      return `/scheme/${schemeId}`;
     },
```

Comment:
more nitpicking: can just use implicit return

```suggestion
    onSuccessRedirectHref: () => `/scheme/${schemeId}`,
```